---
title: "Class15: Transcriptomics and the analysis of RNA-Seq data"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##Import countData and colData into R
```{r}
counts <- read.csv("airway_scaledcounts.csv", stringsAsFactors = FALSE)
metadata <-  read.csv("airway_metadata.csv", stringsAsFactors = FALSE)
head(counts)
head(metadata)
```

```{r}
colnames(counts)
```

```{r}
metadata$id
```

check for corespondence of metadata and countdata
```{r}
all(colnames(counts)[-1] == metadata$id)
```


First task find which columns in **counts** correspond to the control conditions (i.e.. no durg!). Then we can calculate the mean count value for each gene. 
```{r}
control.inds <- metadata$dex == "control"
control.ids <- metadata$id[control.inds]
control.mean <- rowSums( counts[ ,control.ids] ) / ncol(counts[,control.ids])
names(control.mean) <- counts$ensgene
```

Do the same for durg treated samples (i.e. **treated** columns)
```{r}
treated.inds <- metadata$dex == "treated"
treated.ids <- metadata$id[treated.inds]
treated.mean <- rowSums( counts[ ,treated.ids] ) / ncol(counts[,treated.ids])
names(treated.mean) <- counts$ensgene
```

Combine the mean count values for control and treated into a new data.frame called meancounts
```{r}
meancounts <- data.frame(control.mean, treated.mean)
colSums(meancounts)
```

Now we can plot the drug treated vs non_drug control count values
```{r}
plot(meancounts$control.mean, meancounts$treated.mean)
```

Put this on a log log plot 
```{r}
plot(meancounts$control.mean, meancounts$treated.mean, log = "xy", xlab = "log control", ylab = "log treated")
```

We can find candidate differentially expressed genes by looking for genes with a large change between control and dex-treated samples. We usually look at the log2 of the fold change, because this has better mathematical properties.
Here we calculate log2foldchange, add it to our meancounts data.frame and inspect the results either with the head()
```{r}
meancounts$log2fc <- log2(meancounts[,"treated.mean"]/meancounts[,"control.mean"])
head(meancounts)
```

Test for remove zeor entries
```{r}
toy <- c(1,2,0,4,0)
which(toy == 0)
```
```{r}
toy <- meancounts[1:6,]
unique(which(toy[,1:2] == 0, arr.ind = TRUE)[,1])
```

Remove the zero entry genes from meancounts to produce **mycounts**
```{r}
zero.vals <- which(meancounts[,1:2]==0, arr.ind=TRUE)
to.rm <- unique(zero.vals[,1])
mycounts <- meancounts[-to.rm,]
head(mycounts)
```


```{r}
up.ind <- mycounts$log2fc > 2
down.ind <- mycounts$log2fc < (-2)
```

```{r}
sum(up.ind)
sum(down.ind)
```

There are `sum(up.ind)` number of up regulated genes and `sum(down.ind)` down regulated genes
```{r}
head(mycounts[up.ind,])
```

##Adding annotation data
```{r}
anno <- read.csv("annotables_grch38.csv")
head(anno)
```

Use the merge() function to add annotation data from ensemble
```{r}
mycounts.annotated <- merge(x = mycounts, y = anno, by.x = "row.names", by.y = "ensgene")
head(mycounts.annotated)
```


##DESeq2 analysis


```{r}
library(DESeq2)
citation("DESeq2")
```

```{r}
dds <- DESeqDataSetFromMatrix(countData=counts, 
                              colData=metadata, 
                              design=~dex, 
                              tidy=TRUE)
dds
```

```{r}
dds <- DESeq(dds)
```

```{r}
res <- results(dds)
head(res)
summary(res)
```

```{r}
res01 <- results(dds, alpha = 0.01)
res01
```

make vocano plot
```{r}
res$sig <- res$padj<0.05
table(res$sig)
```

```{r}
mycols <- rep("grey", nrow(res01))
mycols[abs(res01$log2FoldChange) > 2] <- "blue"
mycols[res01$padj < 0.01 & abs(res01$log2FoldChange) > 2] <- "red"
plot(res$log2FoldChange, -log(res01$padj), col = mycols)
abline(v = c(-2,2), col = "grey", lty = 2)
abline(h = -log(0,1), col = "grey", lty = 2)
```













